### ISSUE TITLE
Projet 4 : Essayer différents types encodages

### ISSUE THREAD

**Issue by gabriel1628** [gabriel1628/Formation-Data-Scientist-OC#16]
Essayer différents types d'encodage des variables catégorielles. <br>
Différents types d'encodage à tester :
- [OneHotEncoder](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html)
- [TargetEncoder](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.TargetEncoder.html#sklearn.preprocessing.TargetEncoder)
- [GLMMEncoder](https://contrib.scikit-learn.org/category_encoders/glmm.html)
- [LeaveOneOutEncoder](https://contrib.scikit-learn.org/category_encoders/leaveoneout.html)
- [MEstimateEncoder](https://contrib.scikit-learn.org/category_encoders/mestimate.html)
- [HashingEncoder](https://contrib.scikit-learn.org/category_encoders/hashing.html)
- [Hierarchical Bayesian Target Encoding](https://www.kaggle.com/code/mmotoki/hierarchical-bayesian-target-encoding/notebook)
- [Binary Encoding](https://contrib.scikit-learn.org/category_encoders/binary#)

Lien utiles :
- https://www.kaggle.com/code/shahules/an-overview-of-encoding-techniques
- https://towardsdatascience.com/6-ways-to-encode-features-for-machine-learning-algorithms-21593f6238b0
- https://medium.com/@kevin.menear/comparing-label-encoding-one-hot-encoding-and-binary-encoding-for-handling-categorical-variables-933544ccbd02
- https://towardsdatascience.com/all-about-categorical-variable-encoding-305f3361fd02
- https://scikit-learn.org/stable/modules/preprocessing.html#encoding-categorical-features
- https://contrib.scikit-learn.org/category_encoders/
